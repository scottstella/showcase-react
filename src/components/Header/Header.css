.header {
  width: 100%;
  background-color: #7d94b5;
  display: flex;
}

.header h1 {
  align-self: center;
}

.header-logo {
  width: 232px;
  height: 107.5px;
}
