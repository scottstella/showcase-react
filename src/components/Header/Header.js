import React from "react";

export default function Header() {
  return (
    <div class=".header">
      <h1>Showcase React</h1>
    </div>
  );
}
