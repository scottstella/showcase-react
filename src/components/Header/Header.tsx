import React from "react";
import { NavLink } from "react-router-dom";
import hearthstoneBanner from "../../assets/images/hearthstone-banner.png";
import Auth from "../Auth";
import "./Header.css";
import "../Auth.css";

const Header: React.FC = () => {
  return (
    <div className="header">
      <div className="header-left">
        <NavLink to="/">
          <img className="header-logo" src={hearthstoneBanner} alt="hearthstone-banner" />
        </NavLink>
        <h1>Deck Tracker</h1>
      </div>
      <div className="header-right">
        <Auth />
      </div>
    </div>
  );
};

export default Header;
