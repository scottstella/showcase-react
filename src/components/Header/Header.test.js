import React from "react";
import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Header from "./Header";
// Wrap component with Router for NavLink
const renderWithRouter = component => {
  return render(React.createElement(BrowserRouter, null, component));
};
describe("Header", () => {
  it("renders without crashing", () => {
    renderWithRouter(React.createElement(Header, null));
    expect(screen.getByText("Deck Tracker")).toBeInTheDocument();
  });
  it("has the correct CSS class", () => {
    const { container } = renderWithRouter(React.createElement(Header, null));
    expect(container.firstChild).toHaveClass("header");
  });
  it("renders the Hearthstone banner image", () => {
    renderWithRouter(React.createElement(Header, null));
    const image = screen.getByAltText("hearthstone-banner");
    expect(image).toBeInTheDocument();
    expect(image).toHaveClass("header-logo");
    expect(image.tagName.toLowerCase()).toBe("img");
  });
  it("contains a NavLink that links to home", () => {
    renderWithRouter(React.createElement(Header, null));
    const link = screen.getByRole("link");
    expect(link).toHaveAttribute("href", "/");
  });
  it("renders the title text", () => {
    renderWithRouter(React.createElement(Header, null));
    const title = screen.getByText("Deck Tracker");
    expect(title).toBeInTheDocument();
    expect(title.tagName.toLowerCase()).toBe("h1");
  });
  it("matches snapshot", () => {
    const { container } = renderWithRouter(React.createElement(Header, null));
    expect(container.firstChild).toMatchSnapshot();
  });
});
