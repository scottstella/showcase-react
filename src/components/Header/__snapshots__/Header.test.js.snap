// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > matches snapshot 1`] = `
<div
  class="header"
>
  <div
    class="header-left"
  >
    <a
      aria-current="page"
      class="active"
      data-discover="true"
      href="/"
    >
      <img
        alt="hearthstone-banner"
        class="header-logo"
        src="/src/assets/images/hearthstone-banner.png"
      />
    </a>
    <h1>
      Deck Tracker
    </h1>
  </div>
  <div
    class="header-right"
  >
    <div
      class="auth-container"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-spinner fa-spin auth-spinner"
        data-icon="spinner"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M304 48a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zm0 416a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zM48 304a48 48 0 1 0 0-96 48 48 0 1 0 0 96zm464-48a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zM142.9 437A48 48 0 1 0 75 369.1 48 48 0 1 0 142.9 437zm0-294.2A48 48 0 1 0 75 75a48 48 0 1 0 67.9 67.9zM369.1 437A48 48 0 1 0 437 369.1 48 48 0 1 0 369.1 437z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</div>
`;
